### DISCUSSION TITLE
I encountered some problems when using this project.

### DISCUSSION THREAD

**Discussion by Nicolas-One** [danny-avila/LibreChat#288] [Q&A] [answered]
Hello author, thank you for your selfless contribution. I deployed your project using docker-compose, but encountered some problems. Today I have git pull to version 0.4.5, re-docker-compose up but still exist, I hope you can help me take a look, thank you.

1、My Bing can be used, but occasionally encounter `An error occurred. Please try again in a few moments. Error message: Sorry, you need to login first to access this service.`. If you switch to other modes such as creative, etc., non-default fast is easy to trigger this error, the project has used `cloudflare`, and turned on `SSL`, access is already `https`.

2、My Sydney cannot be used, the error message is `An error occurred. Please try again in a few moments. Error message: Sorry, you need to login first to access this service.` or `[Error: The moderation filter triggered. Try again with different wording.]`

3、My ChatGPT cannot be used overall error `An error occurred. Please try again in a few moments. Error message: Failed to send message. HTTP 401 - {"detail":{"message":"Incorrect API key provided: \"eyJhbGc*************YTg\". You can find your API key at https://platform.openai.com/account/api-keys.","type":"invalid_request_error","param":null,"code":"invalid_api_key"}}`, I did not fill in the value of `OPENAI_KEY`, he is still empty, but I see the log shows that it called the browser solution, the log is as follows:

```
chat-clone     | ask log {
chat-clone     |   userMessage: {
chat-clone     |     messageId: 'd328b0*******0758926c4cc',
chat-clone     |     sender: 'User',
chat-clone     |     text: 'hello',
chat-clone     |     parentMessageId: '00000000-0000-0000-0000-000000000000',
chat-clone     |     conversationId: 'b71e*********9fd624a74cf',
chat-clone     |     isCreatedByUser: true
chat-clone     |   },
chat-clone     |   endpointOption: { model: 'text-davinci-002-render-sha', token: null },
chat-clone     |   conversationId: 'b71e0610-f3e0-4c61-a4a7-29fd624a74cf'
chat-clone     | }
chat-meilisearch | [2023-05-17T01:45:23Z INFO  actix_web::middleware::logger] 172.18.0.4 "POST /indexes/messages/documents HTTP/1.1" 202 137 "-" "undici" 0.003084
chat-clone     | gptBrowser clientOptions {
chat-clone     |   reverseProxyUrl: 'https://ai.fakeopen.com/api/conversation',
chat-clone     |   accessToken: '"eyJhbGciOiJSU****************Pwa_rMIsa0xLYTg"',
chat-clone     |   model: 'text-davinci-002-render-sha',
chat-clone     |   debug: false,
chat-clone     |   proxy: null,
chat-clone     |   user: '64632f*****4ddd88'
chat-clone     | }
```
My server is Oracle's Singapore server, has brushed warp's IPV6 IP, and set IPV6 priority, can access `chat.openai.com` through the browser, and by default used `https://ai.fakeopen.com/api/conversation` reverse proxy, I have another project on this server, he also used this reverse proxy, did not encounter any problems, has been used for more than 2 months, but your project encountered problems, do not know how to solve, hope you can help me take a look, thank you.

**Comment by danny-avila** [comment]
What does your api/.env file look like before using docker-compose? Have you deleted the old image before running docker compose? You can try editing the compose file to pull from dockerhub

**Reply by Nicolas-One** [reply]
Thank you for helping me check the issue. The `api/.env` file was copied from `api/.env.example`, and I only modified the values of `BINGAI_TOKEN` and `CHATGPT_TOKEN`, without changing anything else.

Just now, I deleted all the containers and images. However, I encountered a problem where a request after logging in resulted in a 401 error.
![image](https://github.com/danny-avila/chatgpt-clone/assets/23067153/de1bc4c5-56b8-407c-a12f-f9d0351afe9d)

Sydney for BingAI is now available, but only two modes, `creative` and `fast`, are functional. Other modes frequently encounter issues and are inaccessible, regardless of whether it is `BingAI` or `Sydney`. The error message is as follows:
`An error occurred. Please try again in a few moments. Error message: Sorry, you need to login first to access this service.`，

Kindly provide the requested information:
![image](https://github.com/danny-avila/chatgpt-clone/assets/23067153/5892764b-1fa7-43a2-907a-1ad82193835e)

Below is the error log you requested:
![image](https://github.com/danny-avila/chatgpt-clone/assets/23067153/53cfc5ba-e936-4fd9-93d0-58a93d645f07)


Despite encountering occasional errors in the logs, the system managed to provide responses to the queries. Please find the error log below：
![image](https://github.com/danny-avila/chatgpt-clone/assets/23067153/29147789-7ed0-4785-ba3f-2ac66b8e8363)

Apologies for the continued inconvenience you are facing with ChatGPT on the web version, which is still encountering errors.
```
An error occurred. Please try again in a few moments. Error message: Failed to send message. HTTP 401 - {"detail":{"message":"Incorrect API key provided: \"eyJhbGc***********************************YTg\". You can find your API key at https://platform.openai.com/account/api-keys.","type":"invalid_request_error","param":null,"code":"invalid_api_key"}}
```

Below is the error log you requested:
![image](https://github.com/danny-avila/chatgpt-clone/assets/23067153/6a165530-b8bb-4778-a481-e30e1294f82b)

**Reply by danny-avila** [reply]
Are you using a proxy? I think most of these issues are IP restrictions and you will need to use one

See here for similar issue: https://github.com/waylaidwanderer/node-chatgpt-api/issues/205

you should try to use that library individually with the same setup as this project uses it, so you can figure out issues related to it.

 You can ignore the convos 401 if your convos are showing up. The other 401 error is related to titling conversations, which is done with openAI API and needs an API key, otherwise can be ignored. There will be a setting to toggle titling in the future.

it’s hard to debug these issues when I can’t reproduce them.

**Reply by Nicolas-One** [reply]
I don't use a proxy on the server, the problem with the ChatGPT web version is mainly because I use `https://github.com/transitive-bullshit/chatgpt-api` on other projects on this server to send requests, and I also used the same reverse proxy of `https://ai.fakeopen.com/api/conversation`, but my project has no problem, but your project has this problem, I think it is estimated that the reverse proxy has not taken effect The reason, I don't have time to find the specific reason recently, if you don't fix it after a while, I can take a look.
But about the BingAI prompt: `An error occurred. Please try again in a few moments. Error message: Sorry, you need to login first to access this service.`, this problem still occurs quite frequently, I hope you can help me .

**Reply by danny-avila** [reply]
I use a reverse proxy almost every day with no issue with gptBrowser. I’ll share my configuration to see if it works later. Also if it works with transitives library, it may be an issue with how the headers are set for that specific reverse Proxy URL. I will test your reverse proxy with that later (although I don’t trust it and putting my token at risk)

**Reply by danny-avila** [reply]
the reverse proxy for chatGPTBrowser works for me

```
PORT=3080
CHATGPT_REVERSE_PROXY=http://localhost:8090/api/conversation
BINGAI_TOKEN=user_provided
CHATGPT_TOKEN=user_provided
CHATGPT_MODELS=text-davinci-002-render-sha,gpt-4,gpt-4-browsing
BROWSER_MODEL=gpt-4
HOST=localhost
SEARCH=TRUE
MONGO_URI=mongodb://localhost:27018/chatgpt-usersys
OPENAI_KEY=myKey

#JWT:
JWT_SECRET_DEV=secret

# Add a secure secret for production if deploying to live domain.
JWT_SECRET_PROD=secret

# Set the expiration delay for the secure cookie with the JWT token
# Delay is in millisecond e.g. 7 days is 1000*60*60*24*7
SESSION_EXPIRY=1000 * 60 * 60 * 24 * 7

# Site URLs:
# Don't forget to set Node env to development in the Server configuration section above
# if you want to run in dev mode
CLIENT_URL_DEV=http://localhost:3090
SERVER_URL_DEV=http://localhost:3080

# Change these values to domain if deploying:
CLIENT_URL_PROD=http://localhost:3080
SERVER_URL_PROD=http://localhost:3080

GOOGLE_CLIENT_ID=someid
GOOGLE_CLIENT_SECRET=somesecret
GOOGLE_CALLBACK_URL=/oauth/google/callback
```

**Reply by Nicolas-One** [reply, marked as answer]
I updated your project today and then I found a problem. It didn't prompt the previous error, but changed to a different error message. I forgot to save the record. The general idea is that my api key has a problem. Then I went to check the log and found that the request information on the log increased a pair of `' '` symbols. I guessed that this was the problem that caused me to have an error. So I removed the `" "` of CHATGPT_TOKEN in .env and I can access the chatgpt web version.
